### PR TITLE
Add support for Go 1.20

### PR DESCRIPTION
**Pull request by nwithan8**
# Description

See title

# Testing

CI updated to build against Go 1.20, test in Go 1.15 - 1.20

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement (fixing a typo, updating readme, renaming a variable name, etc.)

### CODE REVIEW

**Comment by nwithan8** [comment]
Not sure why the coverage tool is failing. Also failing locally for me.

**Comment by Justintime50** [comment]
Maybe it doesn't yet support 1.20? Go is supposed to be forever backwards compatible but that guarantee may not always be true, especially for tooling?

**Comment by nwithan8** [comment]
> Maybe it doesn't yet support 1.20? Go is supposed to be forever backwards compatible but that guarantee may not always be true, especially for tooling?

Seems to be the case. I retooled our CI so unit tests/build is run on each Go version, but coverage is run separately, on Go 1.19. Also helps us be more obvious with what each CI step is actually testing

**Review by jchen293 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -37,18 +37,43 @@ jobs:
...
       - name: Coveralls
-        if: github.ref == 'refs/heads/master'
         env:
           COVERALLS_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: goveralls -coverprofile=covprofile -service=github
-      - name: Run Gosec
-        uses: securego/gosec@master
-        with:
-          args: -tests -exclude-dir=examples ./...
+        run: goveralls -coverprofile=covprofile -service=github
```
Comment (Justintime50) [inline .github/workflows/ci.yml:79]: nit: newline

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**